sympy/polys/tests/test_polytools.py

```diff
@@ -2881,7 +2882,7 @@ def test_cancel():
     M = MatrixSymbol('M', 5, 5)
     assert cancel(M[0,0] + 7) == M[0,0] + 7
     expr = sin(M[1, 4] + M[2, 1] * 5 * M[4, 0]) - 5 * M[1, 2] / z
-    assert cancel(expr) == expr
+    assert simplify(cancel(expr)) == expr
```

Comment:
`(z*sin(M[1, 4] + M[2, 1] * 5 * M[4, 0]) - 5 * M[1, 2]) / z` is the right answer for cancel to return. The test should be

```py
assert cancel(expr) == (z*sin(M[1, 4] + M[2, 1] * 5 * M[4, 0]) - 5 * M[1, 2]) / z
```

Calling `simplify` in the function that tests the output of `cancel` defeats the purpose.